diff -dur systemd-218.orig/tmpfiles.d/legacy.conf systemd-218/tmpfiles.d/legacy.conf
--- systemd-218.orig/tmpfiles.d/legacy.conf	2014-06-11 00:11:32.000000000 +0200
+++ systemd-218/tmpfiles.d/legacy.conf	2015-04-22 13:11:28.000000000 +0200
@@ -10,13 +10,13 @@
 # These files are considered legacy and are unnecessary on legacy-free
 # systems.
 
-d /run/lock 0755 root root -
+d /run/lock 1771 root root -
 L /var/lock - - - - ../run/lock
 
 # /run/lock/subsys is used for serializing SysV service execution, and
 # hence without use on SysV-less systems.
 
-d /run/lock/subsys 0755 root root -
+d /run/lock/subsys 0700 root root -
 
 # /run/lock/lockdev is used to serialize access to tty devices via
 # LCK..xxx style lock files, For more information see:
@@ -24,7 +24,7 @@
 # On modern systems a BSD file lock is a better choice if
 # serialization is needed on those devices.
 
-d /run/lock/lockdev 0775 root lock -
+# d /run/lock/lockdev 0775 root lock -
 
 # /forcefsck, /fastboot and /forcequotecheck are deprecated in favor of the
 # kernel command line options 'fsck.mode=force', 'fsck.mode=skip' and
