--- systemd-stable-250/tmpfiles.d/legacy.conf.in.orig	2021-12-27 11:37:45.454887047 +0100
+++ systemd-stable-250/tmpfiles.d/legacy.conf.in	2021-12-27 11:39:02.180866644 +0100
@@ -10,7 +10,7 @@
 # by newer interfaces. Their use is discouraged on legacy-free systems. This
 # configuration is provided to maintain backward compatibility.
 
-d /run/lock 0755 root root -
+d /run/lock 1771 root root -
 L /var/lock - - - - ../run/lock
 {% if CREATE_LOG_DIRS %}
 L$ /var/log/README - - - - ../..{{DOC_DIR}}/README.logs
@@ -19,5 +19,5 @@
 {% if HAVE_SYSV_COMPAT %}
 # /run/lock/subsys is used for serializing SysV service execution, and
 # hence without use on SysV-less systems.
-d /run/lock/subsys 0755 root root -
+d /run/lock/subsys 0700 root root -
 {% endif %}
