--- systemd-44/tmpfiles.d/x11.conf~	2011-08-29 16:32:03.000000000 +0200
+++ systemd-44/tmpfiles.d/x11.conf	2012-03-29 11:07:45.026154271 +0200
@@ -8,11 +8,11 @@
 # See tmpfiles.d(5) for details
 
 # Make sure these are created by default so that nobody else can
-d /tmp/.X11-unix 1777 root root 10d
-d /tmp/.ICE-unix 1777 root root 10d
-d /tmp/.XIM-unix 1777 root root 10d
-d /tmp/.font-unix 1777 root root 10d
-d /tmp/.Test-unix 1777 root root 10d
+d /tmp/.X11-unix 1777 root root -
+d /tmp/.ICE-unix 1777 root root -
+d /tmp/.XIM-unix 1777 root root -
+d /tmp/.font-unix 1777 root root -
+d /tmp/.Test-unix 1777 root root -
 
 # Unlink the X11 lock files
 r /tmp/.X[0-9]*-lock
