--- systemd-242/src/gpt-auto-generator/gpt-auto-generator.c.orig	2019-04-11 19:40:01.816893823 +0200
+++ systemd-242/src/gpt-auto-generator/gpt-auto-generator.c	2019-04-11 19:42:53.022916795 +0200
@@ -438,7 +438,7 @@
 
                 /* Use /boot as fallback, but only if there's no XBOOTLDR partition */
                 if (!has_xbootldr) {
-                        esp_path = "/boot";
+                        esp_path = "/boot/efi";
                         id = "boot";
                 }
         }
