--- a/src/gpt-auto-generator/gpt-auto-generator.c~	2016-11-03 18:16:42.000000000 +0100
+++ b/src/gpt-auto-generator/gpt-auto-generator.c	2016-11-12 20:01:04.784509883 +0100
@@ -466,7 +466,7 @@ static int add_boot(const char *what) {
         }
 
         /* If /efi exists we'll use that. Otherwise we'll use /boot, as that's usually the better choice */
-        esp = access("/efi/", F_OK) >= 0 ? "/efi" : "/boot";
+        esp = access("/efi/", F_OK) >= 0 ? "/efi" : "/boot/efi";
 
         /* We create an .automount which is not overridden by the .mount from the fstab generator. */
         if (fstab_is_mount_point(esp)) {
