--- systemd-215/src/efi-boot-generator/efi-boot-generator.c.orig	2014-03-18 18:05:49.801378306 +0100
+++ systemd-215/src/efi-boot-generator/efi-boot-generator.c	2014-07-11 11:44:46.017114762 +0200
@@ -69,9 +69,9 @@ int main(int argc, char *argv[]) {
                 return EXIT_SUCCESS;
         }
 
-        if (path_is_mount_point("/boot", true) <= 0 &&
-            dir_is_empty("/boot") <= 0) {
-                log_debug("/boot already populated, exiting.");
+        if (path_is_mount_point("/boot/efi", true) <= 0 &&
+            dir_is_empty("/boot/efi") <= 0) {
+                log_debug("/boot/efi already populated, exiting.");
                 return EXIT_SUCCESS;
         }
 
@@ -84,7 +84,7 @@ int main(int argc, char *argv[]) {
                 return EXIT_FAILURE;
         }
 
-        name = strappenda(arg_dest, "/boot.mount");
+        name = strappenda(arg_dest, "/boot-efi.mount");
         f = fopen(name, "wxe");
         if (!f) {
                 log_error("Failed to create mount unit file %s: %m", name);
@@ -105,7 +105,7 @@ int main(int argc, char *argv[]) {
                 "Description=EFI System Partition\n"
                 "Documentation=man:systemd-efi-boot-generator(8)\n");
 
-        r = generator_write_fsck_deps(f, arg_dest, what, "/boot", "vfat");
+        r = generator_write_fsck_deps(f, arg_dest, what, "/boot/efi", "vfat");
         if (r < 0)
                 return EXIT_FAILURE;
 
@@ -113,7 +113,7 @@ int main(int argc, char *argv[]) {
                 "\n"
                 "[Mount]\n"
                 "What=%s\n"
-                "Where=/boot\n"
+                "Where=/boot/efi\n"
                 "Type=vfat\n"
                 "Options=umask=0077,noauto\n",
                 what);
@@ -124,7 +124,7 @@ int main(int argc, char *argv[]) {
                 return EXIT_FAILURE;
         }
 
-        name = strappenda(arg_dest, "/boot.automount");
+        name = strappenda(arg_dest, "/boot-efi.automount");
         fclose(f);
         f = fopen(name, "wxe");
         if (!f) {
@@ -136,7 +136,7 @@ int main(int argc, char *argv[]) {
               "[Unit]\n"
               "Description=EFI System Partition Automount\n\n"
               "[Automount]\n"
-              "Where=/boot\n", f);
+              "Where=/boot/efi\n", f);
 
         fflush(f);
         if (ferror(f)) {
@@ -144,10 +144,10 @@ int main(int argc, char *argv[]) {
                 return EXIT_FAILURE;
         }
 
-        name = strappenda(arg_dest, "/" SPECIAL_LOCAL_FS_TARGET ".wants/boot.automount");
+        name = strappenda(arg_dest, "/" SPECIAL_LOCAL_FS_TARGET ".wants/boot-efi.automount");
         mkdir_parents(name, 0755);
 
-        if (symlink("../boot.automount", name) < 0) {
+        if (symlink("../boot-efi.automount", name) < 0) {
                 log_error("Failed to create symlink %s: %m", name);
                 return EXIT_FAILURE;
         }
