--- systemd-stable-245.6/src/login/systemd-user.m4.orig	2020-06-11 13:54:19.219560705 +0200
+++ systemd-stable-245.6/src/login/systemd-user.m4	2020-06-11 13:59:30.489083928 +0200
@@ -2,19 +2,10 @@
 #
 # Used by systemd --user instances.
 
-m4_ifdef(`ENABLE_HOMED',
--account sufficient pam_systemd_home.so
-)m4_dnl
-account sufficient pam_unix.so
-account required pam_permit.so
-
+account		include		system-auth
 m4_ifdef(`HAVE_SELINUX',
-session required pam_selinux.so close
-session required pam_selinux.so nottys open
-)m4_dnl
-session required pam_loginuid.so
-session optional pam_keyinit.so force revoke
-m4_ifdef(`ENABLE_HOMED',
--session optional pam_systemd_home.so
+session		optional	pam_selinux.so close
+session		optional	pam_selinux.so nottys open
 )m4_dnl
-session optional pam_systemd.so
+session		required	pam_loginuid.so
+session		include	system-auth
