--- systemd-38/src/core/service.c~	2012-01-26 00:12:06.239561034 +0100
+++ systemd-38/src/core/service.c	2012-01-26 18:26:15.476825041 +0100
@@ -516,6 +516,13 @@
                 line++;
 
                 t = strstrip(l);
+                // PLD specific ugly hack to detect SysV services requiring network
+                if (startswith_no_case(t, ". /etc/sysconfig/network")) {
+                        r = unit_add_dependency_by_name(UNIT(s), UNIT_REQUIRES | UNIT_AFTER, "network.target", NULL, true);
+
+                        if (r < 0)
+                                 log_error("[%s:%u] Failed to add dependency on network.target, ignoring: %s", path, line, strerror(-r));
+                }
                 if (*t != '#')
                         continue;
 
--- systemd-44/man/systemd.special.xml~	2012-02-20 16:33:32.000000000 +0200
+++ systemd-44/man/systemd.special.xml	2012-03-18 10:44:01.469494195 +0200
@@ -324,6 +324,10 @@
                                         referring to the
                                         <literal>$network</literal>
                                         facility.</para>
+
+										<para>PLD specific: also scripts with
+										<literal>. /etc/sysconfig/network</literal>
+										source line get the dependency.</para>
                                 </listitem>
                         </varlistentry>
                         <varlistentry>
