--- systemd-stable-254.1/src/gpt-auto-generator/gpt-auto-generator.c.orig	2023-08-09 22:33:59.978475054 +0200
+++ systemd-stable-254.1/src/gpt-auto-generator/gpt-auto-generator.c	2023-08-09 22:41:09.618936068 +0200
@@ -582,15 +609,15 @@
         if (r > 0)
                 return 0;
 
-        /* If XBOOTLDR partition is not present and /boot/ is unused and empty, we'll take that.
+        /* If XBOOTLDR partition is not present and /boot/efi/ is unused and empty, we'll take that.
          * Otherwise, if /efi/ is unused and empty (or missing), we'll take that.
          * Otherwise, we do nothing. */
         if (!has_xbootldr) {
-                r = path_is_busy("/boot");
+                r = path_is_busy("/boot/efi");
                 if (r < 0)
                         return r;
                 if (r == 0) {
-                        esp_path = "/boot";
+                        esp_path = "/boot/efi";
                         id = "boot";
                 }
         }
