--- systemd-240/units/systemd-logind.service.in.orig	2019-01-13 10:40:04.142278844 +0100
+++ systemd-240/units/systemd-logind.service.in	2019-01-13 10:41:12.495611397 +0100
@@ -41,3 +41,7 @@
 # Increase the default a bit in order to allow many simultaneous logins since
 # we keep one fd open per session.
 LimitNOFILE=@HIGH_RLIMIT_NOFILE@
+
+# Run as group 'proc' to ensure access to /proc when
+# it is mounted with hidepid=1|2 option
+Group=17
