diff -dur systemd-219.orig/src/python-systemd/_daemon.c systemd-219/src/python-systemd/_daemon.c
--- systemd-219.orig/src/python-systemd/_daemon.c	2015-02-03 02:00:06.000000000 +0100
+++ systemd-219/src/python-systemd/_daemon.c	2015-08-07 12:32:34.000000000 +0200
@@ -143,9 +143,12 @@
         const char *path = NULL;
 
 #if PY_MAJOR_VERSION >=3 && PY_MINOR_VERSION >= 1
+        _cleanup_Py_DECREF_ PyObject *_path = NULL;
         if (!PyArg_ParseTuple(args, "i|O&:_is_fifo",
-                              &fd, Unicode_FSConverter, &path))
+                              &fd, Unicode_FSConverter, &_path))
                 return NULL;
+	if (_path)
+		path = PyBytes_AsString(_path);
 #else
         if (!PyArg_ParseTuple(args, "i|z:_is_fifo", &fd, &path))
                 return NULL;
@@ -171,9 +174,12 @@
         const char *path = NULL;
 
 #if PY_MAJOR_VERSION >=3 && PY_MINOR_VERSION >= 1
+        _cleanup_Py_DECREF_ PyObject *_path = NULL;
         if (!PyArg_ParseTuple(args, "i|O&:_is_mq",
-                              &fd, Unicode_FSConverter, &path))
+                              &fd, Unicode_FSConverter, &_path))
                 return NULL;
+	if (_path)
+		path = PyBytes_AsString(_path);
 #else
         if (!PyArg_ParseTuple(args, "i|z:_is_mq", &fd, &path))
                 return NULL;
