--- systemd-220/src/shared/generator.c~	2015-05-15 17:06:10.000000000 +0200
+++ systemd-220/src/shared/generator.c	2015-05-25 21:21:02.031196707 +0200
@@ -61,7 +61,7 @@
                 "[Service]\n"
                 "Type=oneshot\n"
                 "RemainAfterExit=yes\n"
-                "ExecStart=/usr/lib/systemd/systemd-fsck %2$s\n"
+                "ExecStart=/lib/systemd/systemd-fsck %2$s\n"
                 "TimeoutSec=0\n",
                 program_invocation_short_name,
                 what,
