--- systemd-233/sysctl.d/50-default.conf~	2017-03-01 22:43:06.000000000 +0100
+++ systemd-233/sysctl.d/50-default.conf	2017-06-28 14:09:04.347320793 +0200
@@ -11,13 +11,6 @@
 # (e.g. /etc/sysctl.d/90-override.conf), and put any assignments
 # there.
 
-# System Request functionality of the kernel (SYNC)
-#
-# Use kernel.sysrq = 1 to allow all keys.
-# See https://www.kernel.org/doc/html/latest/admin-guide/sysrq.html for a list
-# of values and keys.
-kernel.sysrq = 16
-
 # Append the PID to the core filename
 kernel.core_uses_pid = 1
 
