From 0664daffee63648ff93b5571421149383501d356 Mon Sep 17 00:00:00 2001
From: Jan Palus <atler@pld-linux.org>
Date: Thu, 11 Jun 2020 18:23:28 +0200
Subject: [PATCH] login: filenames in /run/systemd/users are uids

Fixes: #16146
Signed-off-by: Jan Palus <atler@pld-linux.org>
---
 src/login/logind.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/login/logind.c b/src/login/logind.c
index 5a556f9ea4..3239f7d18a 100644
--- a/src/login/logind.c
+++ b/src/login/logind.c
@@ -328,11 +328,18 @@ static int manager_enumerate_users(Manager *m) {
 
         FOREACH_DIRENT(de, d, return -errno) {
                 User *u;
+                uid_t uid;
 
                 if (!dirent_is_file(de))
                         continue;
 
-                k = manager_add_user_by_name(m, de->d_name, &u);
+                k = safe_atou(de->d_name, &uid);
+                if (k < 0) {
+                        r = log_warning_errno(k, "Failed to add user by file name %s, ignoring: %m", de->d_name);
+                        continue;
+                }
+
+                k = manager_add_user_by_uid(m, uid, &u);
                 if (k < 0) {
                         r = log_warning_errno(k, "Failed to add user by file name %s, ignoring: %m", de->d_name);
                         continue;
-- 
2.27.0

