--- systemd-stable-254/units/meson.build.orig	2023-08-03 00:34:19.694943339 +0200
+++ systemd-stable-254/units/meson.build	2023-08-03 00:37:00.467127234 +0200
@@ -208,6 +208,7 @@
           'file' : 'sys-kernel-config.mount',
           'symlinks' : ['sysinit.target.wants/'],
         },
+        { 'file' : 'sys-kernel-config.service' },
         {
           'file' : 'sys-kernel-debug.mount',
           'symlinks' : ['sysinit.target.wants/'],
diff -durN -x '*.orig' systemd-208.orig/units/sys-kernel-config.mount systemd-208/units/sys-kernel-config.mount
--- systemd-208.orig/units/sys-kernel-config.mount	2013-08-13 22:02:52.000000000 +0200
+++ systemd-208/units/sys-kernel-config.mount	2014-01-10 10:22:04.000000000 +0100
@@ -10,8 +10,9 @@ Description=Configuration File System
 Documentation=https://docs.kernel.org/filesystems/configfs.html
 Documentation=https://www.freedesktop.org/wiki/Software/systemd/APIFileSystems
 DefaultDependencies=no
-ConditionPathExists=/sys/kernel/config
 ConditionCapability=CAP_SYS_RAWIO
+Wants=sys-kernel-config.service
+After=systemd-modules-load.service sys-kernel-config.service
 Before=sysinit.target
 
 # These dependencies are used to make certain that the module is fully
diff -durN -x '*.orig' systemd-208.orig/units/sys-kernel-config.service systemd-208/units/sys-kernel-config.service
--- systemd-208.orig/units/sys-kernel-config.service	1970-01-01 01:00:00.000000000 +0100
+++ systemd-208/units/sys-kernel-config.service	2014-01-10 10:22:04.000000000 +0100
@@ -0,0 +1,8 @@
+[Unit]
+Description=Load the configfs module for /sys/kernel/config mount
+ConditionPathExists=!/sys/kernel/config
+DefaultDependencies=no
+
+[Service]
+Type=oneshot
+ExecStart=/sbin/modprobe -b configfs
