diff -durN systemd-208.orig/Makefile.am systemd-208/Makefile.am
--- systemd-208.orig/Makefile.am	2014-01-06 11:46:16.758882910 +0100
+++ systemd-208/Makefile.am	2014-01-06 11:45:46.942216959 +0100
@@ -411,6 +411,7 @@
 	units/syslog.socket \
 	units/dev-hugepages.mount \
 	units/dev-mqueue.mount \
+	units/sys-kernel-config.service \
 	units/sys-kernel-config.mount \
 	units/sys-kernel-debug.mount \
 	units/sys-fs-fuse-connections.mount \
diff -durN systemd-208.orig/units/sys-kernel-config.mount systemd-208/units/sys-kernel-config.mount
--- systemd-208.orig/units/sys-kernel-config.mount	2013-08-13 22:02:52.798756199 +0200
+++ systemd-208/units/sys-kernel-config.mount	2014-01-06 11:42:49.095554041 +0100
@@ -10,8 +10,8 @@
 Documentation=https://www.kernel.org/doc/Documentation/filesystems/configfs/configfs.txt
 Documentation=http://www.freedesktop.org/wiki/Software/systemd/APIFileSystems
 DefaultDependencies=no
-ConditionPathExists=/sys/kernel/config
-After=systemd-modules-load.service
+Wants=sys-kernel-config.service
+After=systemd-modules-load.service sys-kernel-config.service
 Before=sysinit.target
 
 [Mount]
diff -durN systemd-208.orig/units/sys-kernel-config.service systemd-208/units/sys-kernel-config.service
--- systemd-208.orig/units/sys-kernel-config.service	1970-01-01 01:00:00.000000000 +0100
+++ systemd-208/units/sys-kernel-config.service	2014-01-06 11:41:04.962223257 +0100
@@ -0,0 +1,8 @@
+[Unit]
+Description=Load the configfs module for /sys/kernel/config mount
+ConditionPathExists=!/sys/kernel/config
+DefaultDependencies=no
+
+[Service]
+Type=oneshot
+ExecStart=/sbin/modprobe -b configfs
