--- systemd-stable-249.4/src/core/systemd.pc.in.orig	2021-08-24 13:04:08.000000000 +0200
+++ systemd-stable-249.4/src/core/systemd.pc.in	2021-09-27 19:10:02.119608114 +0200
@@ -65,16 +65,16 @@ systemdshutdowndir=${systemd_shutdown_di
 
 user_tmpfiles_dir=${prefix}/share/user-tmpfiles.d
 
-sysusers_dir=${rootprefix}/lib/sysusers.d
+sysusers_dir=${prefix}/lib/sysusers.d
 sysusersdir=${sysusers_dir}
 
-sysctl_dir=${rootprefix}/lib/sysctl.d
+sysctl_dir=${prefix}/lib/sysctl.d
 sysctldir=${sysctl_dir}
 
-binfmt_dir=${rootprefix}/lib/binfmt.d
+binfmt_dir=${prefix}/lib/binfmt.d
 binfmtdir=${binfmt_dir}
 
-modules_load_dir=${rootprefix}/lib/modules-load.d
+modules_load_dir=${prefix}/lib/modules-load.d
 modulesloaddir=${modules_load_dir}
 
 catalog_dir=${prefix}/lib/systemd/catalog
