diff -urN systemd-241.orig/shell-completion/bash/meson.build systemd-241/shell-completion/bash/meson.build
--- systemd-241.orig/shell-completion/bash/meson.build	2019-02-14 11:11:58.000000000 +0100
+++ systemd-241/shell-completion/bash/meson.build	2019-03-07 01:08:55.312337719 +0100
@@ -30,7 +30,7 @@
          ['systemd-path',        ''],
          ['systemd-run',         ''],
          ['udevadm',             ''],
-         ['bootctl',             'ENABLE_BOOTLOADER'],
+         ['bootctl',             ''],
          ['coredumpctl',         'ENABLE_COREDUMP'],
          ['homectl',             'ENABLE_HOMED'],
          ['hostnamectl',         'ENABLE_HOSTNAMED'],
diff -urN systemd-241.orig/shell-completion/zsh/meson.build systemd-241/shell-completion/zsh/meson.build
--- systemd-241.orig/shell-completion/zsh/meson.build	2019-02-14 11:11:58.000000000 +0100
+++ systemd-241/shell-completion/zsh/meson.build	2019-03-07 01:09:00.655641470 +0100
@@ -25,7 +25,7 @@
          ['_sd_outputmodes',           ''],
          ['_sd_unit_files',            ''],
          ['_sd_machines',              ''],
-         ['_bootctl',                  'ENABLE_BOOTLOADER'],
+         ['_bootctl',                  ''],
          ['_coredumpctl',              'ENABLE_COREDUMP'],
          ['_hostnamectl',              'ENABLE_HOSTNAMED'],
          ['_localectl',                'ENABLE_LOCALED'],
