--- systemd-235/units/meson.build.org	2017-10-21 15:20:03.280741248 +0200
+++ systemd-235/units/meson.build	2017-10-21 15:20:23.741288778 +0200
@@ -76,6 +76,7 @@ units = [
          'sysinit.target.wants/'],
         ['sys-kernel-config.mount',             '',
          'sysinit.target.wants/'],
+        ['sys-kernel-config.service',           ''],
         ['sys-kernel-debug.mount',              '',
          'sysinit.target.wants/'],
         ['sys-kernel-tracing.mount',            '',
diff -durN -x '*.orig' systemd-208.orig/units/sys-kernel-config.mount systemd-208/units/sys-kernel-config.mount
--- systemd-208.orig/units/sys-kernel-config.mount	2013-08-13 22:02:52.000000000 +0200
+++ systemd-208/units/sys-kernel-config.mount	2014-01-10 10:22:04.000000000 +0100
@@ -12,7 +12,8 @@ Description=Kernel Configuration File Sy
 Documentation=https://www.kernel.org/doc/Documentation/filesystems/configfs/configfs.txt
 Documentation=https://www.freedesktop.org/wiki/Software/systemd/APIFileSystems
 DefaultDependencies=no
-ConditionPathExists=/sys/kernel/config
+Wants=sys-kernel-config.service
+After=systemd-modules-load.service sys-kernel-config.service
 ConditionCapability=CAP_SYS_RAWIO
 Before=sysinit.target
 
diff -durN -x '*.orig' systemd-208.orig/units/sys-kernel-config.service systemd-208/units/sys-kernel-config.service
--- systemd-208.orig/units/sys-kernel-config.service	1970-01-01 01:00:00.000000000 +0100
+++ systemd-208/units/sys-kernel-config.service	2014-01-10 10:22:04.000000000 +0100
@@ -0,0 +1,8 @@
+[Unit]
+Description=Load the configfs module for /sys/kernel/config mount
+ConditionPathExists=!/sys/kernel/config
+DefaultDependencies=no
+
+[Service]
+Type=oneshot
+ExecStart=/sbin/modprobe -b configfs
