--- systemd-198/units/initrd-cleanup.service~	2013-03-07 13:07:09.946850823 +0100
+++ systemd-198/units/initrd-cleanup.service	2013-03-21 14:50:00.400052081 +0100
@@ -15,4 +15,4 @@
 
 [Service]
 Type=oneshot
-ExecStart=/usr/bin/systemctl --no-block isolate initrd-switch-root.target
+ExecStart=@SYSTEMCTL@ --no-block isolate initrd-switch-root.target
--- systemd-198/units/initrd-parse-etc.service~	2013-03-07 13:07:09.946850823 +0100
+++ systemd-198/units/initrd-parse-etc.service	2013-03-21 14:50:59.820556209 +0100
@@ -15,6 +15,6 @@
 
 [Service]
 Type=oneshot
-ExecStartPre=/usr/bin/systemctl daemon-reload
-ExecStart=/usr/bin/systemctl start local-fs.target
-ExecStart=/usr/bin/systemctl --no-block start initrd-cleanup.service
+ExecStartPre=@SYSTEMCTL@ daemon-reload
+ExecStart=@SYSTEMCTL@ start local-fs.target
+ExecStart=@SYSTEMCTL@ --no-block start initrd-cleanup.service
--- systemd-198/units/initrd-switch-root.service~	2013-03-05 00:43:05.950383752 +0100
+++ systemd-198/units/initrd-switch-root.service	2013-03-21 14:51:14.987351504 +0100
@@ -15,5 +15,5 @@
 [Service]
 Type=oneshot
 # we have to use "--force" here, otherwise systemd would umount /run
-ExecStart=/usr/bin/systemctl --no-block --force switch-root /sysroot
+ExecStart=@SYSTEMCTL@ --no-block --force switch-root /sysroot
 KillMode=none
