--- systemd-26/configure.ac.orig	2011-05-02 14:05:34.544253875 +0200
+++ systemd-26/configure.ac	2011-05-02 14:22:04.085559738 +0200
@@ -369,6 +369,12 @@ case $with_distro in
                 M4_DISTRO_FLAG=-DTARGET_MANDRIVA=1
 		have_plymouth=true
                 ;;
+        pld)
+    		SYSTEM_SYSVINIT_PATH=/etc/rc.d/init.d
+    		SYSTEM_SYSVRCND_PATH=/etc/rc.d
+    		AC_DEFINE(TARGET_PLD, [], [Target is PLD Linux])
+    		M4_DISTRO_FLAG=-DTARGET_PLD=1
+    		;;
         meego)
                 SYSTEM_SYSVINIT_PATH=
                 SYSTEM_SYSVRCND_PATH=
