diff -dur -x '*~' systemd-219.orig/units/systemd-timesyncd.service.in systemd-219/units/systemd-timesyncd.service.in
--- systemd-219.orig/units/systemd-timesyncd.service.in	2015-02-11 18:28:06.000000000 +0100
+++ systemd-219/units/systemd-timesyncd.service.in	2015-07-10 09:45:32.000000000 +0200
@@ -9,7 +9,7 @@
 Description=Network Time Synchronization
 Documentation=man:systemd-timesyncd.service(8)
 ConditionCapability=CAP_SYS_TIME
-ConditionVirtualization=no
+ConditionVirtualization=!container
 DefaultDependencies=no
 RequiresMountsFor=/var/lib/systemd/clock
 After=systemd-remount-fs.service systemd-tmpfiles-setup.service systemd-sysusers.service
