--- systemd-39/units/systemd-tmpfiles-setup.service.in~	2011-12-30 22:27:36.623529054 +0100
+++ systemd-39/units/systemd-tmpfiles-setup.service.in	2012-01-29 22:55:46.116130563 +0100
@@ -19,4 +19,4 @@
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=@rootbindir@/systemd-tmpfiles --create --remove
+ExecStart=-@rootbindir@/systemd-tmpfiles --create --remove
--- systemd-39/units/systemd-tmpfiles-clean.service.in~	2011-12-30 22:27:36.520529020 +0100
+++ systemd-39/units/systemd-tmpfiles-clean.service.in	2012-01-29 22:56:15.626015474 +0100
@@ -18,5 +18,5 @@
 
 [Service]
 Type=oneshot
-ExecStart=@rootbindir@/systemd-tmpfiles --clean
+ExecStart=-@rootbindir@/systemd-tmpfiles --clean
 IOSchedulingClass=idle
