--- systemd-183/Makefile.am~	2012-05-28 12:46:43.097747775 +0200
+++ systemd-183/Makefile.am	2012-05-28 13:02:55.879128360 +0200
@@ -70,8 +70,8 @@
 systemsleepdir=$(rootlibexecdir)/system-sleep
 systemunitdir=$(rootprefix)/lib/systemd/system
 udevlibexecdir=$(rootprefix)/lib/udev
-udevhomedir = $(libexecdir)/udev
-udevrulesdir = $(libexecdir)/udev/rules.d
+udevhomedir=$(udevlibexecdir)
+udevrulesdir=$(udevlibexecdir)/rules.d
 
 # And these are the special ones for /
 rootprefix=@rootprefix@
@@ -131,7 +131,7 @@
 	-DSYSTEM_SHUTDOWN_PATH=\"$(systemshutdowndir)\" \
 	-DSYSTEMD_KBD_MODEL_MAP=\"$(pkgdatadir)/kbd-model-map\" \
 	-DX_SERVER=\"$(bindir)/X\" \
-	-DUDEVLIBEXECDIR=\""$(libexecdir)/udev"\" \
+	-DUDEVLIBEXECDIR=\""$(udevlibexecdir)"\" \
 	-DPOLKIT_AGENT_BINARY_PATH=\"$(bindir)/pkttyagent\" \
 	-I $(top_srcdir)/src \
 	-I $(top_srcdir)/src/shared \
@@ -1434,7 +1434,7 @@
 
 udev-confdirs:
 	-mkdir -p $(DESTDIR)$(sysconfdir)/udev/rules.d
-	-mkdir -p $(DESTDIR)$(libexecdir)/udev/devices
+	-mkdir -p $(DESTDIR)$(udevlibexecdir)/devices
 
 INSTALL_DATA_HOOKS += udev-confdirs
 
@@ -1906,7 +1906,8 @@
 	src/udev/keymap/check-keymaps.sh
 
 EXTRA_DIST += \
-	src/udev/keymap/check-keymaps.sh
+	src/udev/keymap/check-keymaps.sh \
+	src/udev/keymap/keyboard-force-release.sh.in
 
 CLEANFILES += \
 	$(nodist_keymap_SOURCES) \
@@ -1913,7 +1913,7 @@
 	src/udev/keymap/keys.txt \
 	src/udev/keymap/keys-from-name.gperf
 
-udevkeymapdir = $(libexecdir)/udev/keymaps
+udevkeymapdir = $(udevlibexecdir)/keymaps
 dist_udevkeymap_DATA = \
 	keymaps/acer \
 	keymaps/acer-aspire_5720 \
@@ -1974,7 +1974,7 @@
 	keymaps/toshiba-satellite_m30x \
 	keymaps/zepto-znote
 
-udevkeymapforcereldir = $(libexecdir)/udev/keymaps/force-release
+udevkeymapforcereldir = $(udevlibexecdir)/keymaps/force-release
 dist_udevkeymapforcerel_DATA = \
 	keymaps-force-release/dell-touchpad \
 	keymaps-force-release/dell-xps \
@@ -2981,7 +2981,7 @@
 		-e 's,@includedir\@,$(includedir),g' \
 		-e 's,@VERSION\@,$(VERSION),g' \
 		-e 's,@rootprefix\@,$(rootprefix),g' \
-		-e 's,@udevlibexecdir\@,$(libexecdir)/udev,g' \
+		-e 's,@udevlibexecdir\@,$(udevlibexecdir),g' \
 		< $< > $@ || rm $@
 
 units/%: units/%.in Makefile
--- systemd-183/src/udev/keymap/keyboard-force-release.sh~	2012-04-10 22:32:29.934866904 +0200
+++ systemd-183/src/udev/keymap/keyboard-force-release.sh	2012-05-29 12:52:33.283726546 +0200
@@ -6,7 +6,7 @@
 
 case "$2" in
         /*) scf="$2" ;;
-        *)  scf="@pkglibexecdir@/keymaps/force-release/$2" ;;
+        *)  scf="@udevlibexecdir@/keymaps/force-release/$2" ;;
 esac
 
 read attr <"/sys/$1/force_release"
--- systemd-183/src/udev/udev.pc.in~	2012-04-05 17:49:14.329482586 +0200
+++ systemd-183/src/udev/udev.pc.in	2012-05-31 12:38:37.835904467 +0200
@@ -2,4 +2,4 @@
 Description: udev
 Version: @VERSION@
 
-udevdir=@pkglibexecdir@
+udevdir=@udevlibexecdir@
