--- systemd-183/Makefile.am~	2012-05-30 14:28:53.647009365 +0200
+++ systemd-183/Makefile.am	2012-05-30 14:35:41.593661644 +0200
@@ -3817,6 +3817,7 @@
        '|rootlibexecdir=$(rootlibexecdir)|' \
        '|rootbindir=$(rootbindir)|' \
        '|bindir=$(bindir)|' \
+       '|sbindir=$(sbindir)|' \
        '|SYSTEMCTL=$(rootbindir)/systemctl|' \
        '|SYSTEMD_NOTIFY=$(rootbindir)/systemd-notify|' \
        '|pkgsysconfdir=$(pkgsysconfdir)|' \
--- systemd-183/units/systemd-udev-trigger.service.in~	2012-05-21 15:14:51.642981269 +0200
+++ systemd-183/units/systemd-udev-trigger.service.in	2012-05-30 13:55:20.229509590 +0200
@@ -17,4 +17,4 @@
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=@rootbindir@/udevadm trigger --type=subsystems --action=add ; @rootbindir@/udevadm trigger --type=devices --action=add
+ExecStart=@sbindir@/udevadm trigger --type=subsystems --action=add ; @sbindir@/udevadm trigger --type=devices --action=add
--- systemd-183/units/systemd-udev-settle.service.in~	2012-05-30 13:43:34.886201138 +0200
+++ systemd-183/units/systemd-udev-settle.service.in	2012-05-30 13:55:44.546175399 +0200
@@ -29,4 +29,4 @@
 Type=oneshot
 TimeoutSec=180
 RemainAfterExit=yes
-ExecStart=@rootbindir@/udevadm settle
+ExecStart=@sbindir@/udevadm settle
--- systemd-199/units/initrd-udevadm-cleanup-db.service.in~	2013-03-11 18:01:33.616138706 +0100
+++ systemd-199/units/initrd-udevadm-cleanup-db.service.in	2013-03-26 19:09:54.525625477 +0100
@@ -15,4 +15,4 @@
 
 [Service]
 Type=oneshot
-ExecStart=-@rootbindir@/udevadm info --cleanup-db
+ExecStart=-@sbindir@/udevadm info --cleanup-db
