diff -ur udev-150/extras/ata_id/ata_id.c udev-150-uclibc/extras/ata_id/ata_id.c
--- udev-150/extras/ata_id/ata_id.c	2009-12-08 18:43:09.000000000 +0100
+++ udev-150-uclibc/extras/ata_id/ata_id.c	2010-01-19 17:10:19.000000000 +0100
@@ -168,7 +168,11 @@
 	uint16_t *p;
 
 	p = (uint16_t *) identify;
+#ifdef __UCLIBC__
+	p[offset_words] = __le16_to_cpu (p[offset_words]);
+#else
 	p[offset_words] = le16toh (p[offset_words]);
+#endif
 }
 
 /**
diff -ur udev-150/udev/udevadm-info.c udev-150-uclibc/udev/udevadm-info.c
--- udev-150/udev/udevadm-info.c	2009-12-24 20:40:34.000000000 +0100
+++ udev-150-uclibc/udev/udevadm-info.c	2010-01-19 17:47:00.000000000 +0100
@@ -203,6 +203,9 @@
 static void cleanup_dir(DIR *dir, mode_t mask, int depth)
 {
 	struct dirent *dent;
+#ifdef __UCLIBC__
+	char filename[UTIL_PATH_SIZE];
+#endif
 
 	if (depth <= 0)
 		return;
@@ -212,7 +215,12 @@
 
 		if (dent->d_name[0] == '.')
 			continue;
+#ifdef __UCLIBC__
+		util_strscpyl(filename, sizeof(filename), udev_device_get_syspath(device), "/", dent->d_name, NULL);
+		if (lstat(filename, &statbuf) != 0)
+#else
 		if (fstatat(dirfd(dir), dent->d_name, &stats, AT_SYMLINK_NOFOLLOW) != 0)
+#endif
 			continue;
 		if ((stats.st_mode & mask) != 0)
 			continue;
