--- systemd-238/sysusers.d/basic.conf.in~	2018-03-05 23:16:37.000000000 +0100
+++ systemd-238/sysusers.d/basic.conf.in	2018-05-15 13:30:41.922818679 +0200
@@ -12,27 +12,27 @@ u root    0     "Super User" /root
 u @NOBODY_USER_NAME@  65534 "Nobody"     -
 
 # Administrator group: can *see* more than normal users
-g adm     -     -            -
+g adm     3     -            -
 
 # Administrator group: can *do* more than normal users
-g wheel   -     -            -
+g wheel   10     -            -
 
 # Access to certain kernel and userspace facilities
-g kmem    -     -            -
+g kmem    9     -            -
 g tty     @TTY_GID@     -            -
-g utmp    -     -            -
+g utmp    22    -            -
 
 # Hardware access groups
-g audio   -     -            -
-g cdrom   -     -            -
-g dialout -     -            -
-g disk    -     -            -
-g input   -     -            -
-g kvm     -     -            -
-g lp      -     -            -
+g audio   23    -            -
+g cdrom   27    -            -
+g dialout 16    -            -
+g disk    6     -            -
+g input   182   -            -
+g kvm     160   -            -
+g lp      7     -            -
 g render  -     -            -
-g tape    -     -            -
-g video   -     -            -
+g tape    68    -            -
+g video   24    -            -
 
 # Default group for normal users
 g users   @USERS_GID@     -            -
--- systemd-238/sysusers.d/systemd-remote.conf.m4~	2018-03-05 23:16:37.000000000 +0100
+++ systemd-238/sysusers.d/systemd-remote.conf.m4	2018-05-15 13:33:04.453829878 +0200
@@ -6,5 +6,5 @@
 #  (at your option) any later version.
 
 m4_ifdef(`HAVE_MICROHTTPD',
-u systemd-journal-remote  - "systemd Journal Remote"
+u systemd-journal-remote  319 "systemd Journal Remote"
 )m4_dnl
--- systemd-238/sysusers.d/systemd.conf.m4~	2018-03-05 23:16:37.000000000 +0100
+++ systemd-238/sysusers.d/systemd.conf.m4	2018-05-15 13:34:07.479084303 +0200
@@ -16,5 +16,5 @@ m4_ifdef(`ENABLE_TIMESYNCD',
 u systemd-timesync  - "systemd Time Synchronization"
 )m4_dnl
 m4_ifdef(`ENABLE_COREDUMP',
-u systemd-coredump  - "systemd Core Dumper"
+u systemd-coredump  333 "systemd Core Dumper"
 )m4_dnl

